cves/CVE-2012-4387.yml

```diff
@@ -192,8 +214,12 @@ lessons:
     applies:
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      Similar to the above point on not trusting inputs, perhaps it would've been
+      wiser from an architectural perspective to make requests not directly translate
```

Comment:
Does this mean that another approach would have been better, or does the team's OGNL code just more thorough input validation?

Reply:
I'm don't think that the choice to use OGNL was necessarily bad, but it's important to understand the inherent limitations and vulnerabilities present in any third-party technology you opt to use in your project. In this case, that means more through input validation.